### PR TITLE
update(page): version 1.11

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- version bump
- DT header corrections (project url, misisng author GitHub handle,
  proper versioning, ditch old TS requirements)
- linting improviements (incremental)
- static `create` method added
- `clickHandler` method added
- `window` configuration option
- tests amended
- maintainer added

https://github.com/visionmedia/page.js/compare/1.8.6...v1.11.6

Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## page ([<kbd>unpkg</kbd>](https://unpkg.com/browse/page@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'page'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. callbacks
1. exits
1. base
1. strict

<details>
<summary>as well as these 16 other properties...</summary>
<p>start, stop, show, back, redirect, replace, dispatch, exit, configure, sameOrigin, clickHandler, create, len, current, Context, Route</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against dc6278d094a390f9b24ffdb94830768d3eaa6398
</p>

File: types/page/index.d.ts [hunk 1]
```diff
@@ -167,6 +169,11 @@ declare namespace PageJS {
          * Equivalent to page.exit('*', callback).
          */
         exit(callback: Callback): void;
+
+        /**
+         * This is the click handler used by page to handle routing when a user clicks an anchor like `<a href="/user/profile">`
+         */
+        clickHandler(e: UIEvent): void;
```
Comment (43081j) [inline types/page/index.d.ts:176]: should probably be a `MouseEvent` since it exists to handle click events

File: types/page/page-tests.ts [hunk 2]
```diff
@@ -53,41 +90,10 @@ page('/', index2);
...
-    img.style.display = 'block';
-    text('Showing ' + ctx.params.name);
-}
-function notfound() {
-    document.querySelector('p')
-    .textContent = 'not found';
-}
+
+declare var iframe: HTMLIFrameElement;
+const otherPage = page.create({
```
Comment (43081j) [inline types/page/page-tests.ts:95]: can you add an extra line to test that you can call it without options: `page.create()`

File: types/page/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/page/index.d.ts:176]: Yes, I've been wondering, but opted more general one, would go 'MouseEvent' way though, thx!

Comment (peterblazejewicz) [inline types/page/index.d.ts:176]:
```suggestion
        clickHandler(e: MouseEvent): void;
```

Comment (peterblazejewicz) [reply types/page/index.d.ts]: Done, thx!

File: types/page/page-tests.ts [hunk 3]
```diff
@@ -53,41 +90,10 @@ page('/', index2);
...
+    window: iframe.contentWindow!,
+});
+otherPage('/', index2);
+otherPage.clickHandler; // $ExpectType (e: UIEvent) => void
```
Comment (peterblazejewicz) [inline types/page/page-tests.ts:99]:
```suggestion
otherPage.clickHandler; // $ExpectType (e: MouseEvent) => void
```

File: types/page/page-tests.ts [hunk 4]
```diff
@@ -53,41 +90,10 @@ page('/', index2);
...
-    img.src = ctx['avatar'];
-    img.style.display = 'block';
-    text('Showing ' + ctx.params.name);
-}
-function notfound() {
-    document.querySelector('p')
-    .textContent = 'not found';
-}
+
+declare var iframe: HTMLIFrameElement;
```
Comment (peterblazejewicz) [inline types/page/page-tests.ts:94]:
```suggestion
const homePage = page.create();
const aboutPage = page.create({});

declare var iframe: HTMLIFrameElement;
```

File: types/page/page-tests.ts (hunk 2 above)
Comment (peterblazejewicz) [reply types/page/page-tests.ts:98]: Done, via PR suggestions tool, thx!

**Review by 43081j — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
@typescript-bot, do you mind?

**Review by amcasey — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
Ready to merge 💋

**Pull request merged**